ArduCopter/ekf_check.cpp

```diff
@@ -160,6 +136,17 @@ void Copter::failsafe_ekf_event()
     failsafe.ekf = true;
     Log_Write_Error(ERROR_SUBSYSTEM_FAILSAFE_EKFINAV, ERROR_CODE_FAILSAFE_OCCURRED);
 
+    // does this mode require position?
+    if (!copter.flightmode->requires_GPS() && (g.fs_ekf_action != FS_EKF_ACTION_LAND_EVEN_STABILIZE)) {
+        return;
+    }
+
+    // sometimes LAND *does* require GPS:
+    if (control_mode == LAND && landing_with_GPS()) {
+        mode_land.do_not_use_GPS();
```

Comment:
it doesn't seem right : landing_with_GPS() == true leads to mode_land.do_not_use_GPS() ?

Reply:
oh no it is right, we are in ekf failsafe action . so that is good

Reply:
In fact, should we not always do landing without GPS in case of EKF failsafe ?

Reply:
Yes, we should always land without GPS in case of EKF failsafe and it does.  This is handled by the Copter::ModeLand::init() method that checks if position_ok() (which checks if we're in ekf failsafe).